### PR TITLE
Weather Story WFO Publish Confirmation

### PR DESCRIPTION
**Pull request by eric-gade**
## What does this PR do? 🛠️
After eons, this PR addresses #1269. The gist is that when creating a new Weather Story and entering a linked WFO, authors should see a dialog of some sort informing them that there is another Weather Story already published with the same WFO. The dialog will have options to save a draft, cancel, or "replace" -- meaning unpublish the other Weather Story first, then publish the new one.

### Hey, isn't there a way to already do this? ###
th
  
**No.**
  
I spent a significant amount of time trying to get this "confirmation, with original form data" concept to work via various Drupal APIs and patterns. None of them worked.
  
After _days_ of poking at it, I threw my hands up and decided to go with Javascript and some basic hook/template alterations.
  
### Ok, sure, but how does it work? ###
We add a custom [JS file](https://github.com/weather-gov/weather.gov/blob/eg-1269-basic-js-ws-confirmation/web/themes/weathergov_admin/assets/js/weather-story-modal.js) as a theme library called `weather-story-modal-dialog` which handles the logic of an otherwise inert `<dialog>` element that is now present in the template for adding weather stories.
  
In the admin theme file, we add a few helper methods and hooks that will load this library.
  
Additionally, if the form is a Weather Story create form, the hook will query the system for all weather stories that are currently published, passing information about all of them -- including and especially i[nformation about their linked WFO Terms](https://github.com/weather-gov/weather.gov/blob/eg-1269-basic-js-ws-confirmation/web/themes/weathergov_admin/weathergov_admin.theme#L196-L217) -- to the template file as [a JSON object tucked away in a script tag](https://github.com/weather-gov/weather.gov/blob/eg-1269-basic-js-ws-confirmation/web/themes/weathergov_admin/templates/page--node--add--weather-story.html.twig#L61-L63).
  
If the user decides to publish anyway, all existing published Weather Stories with the same WFO will be archived. This happens [by default in the publish action now](https://github.com/weather-gov/weather.gov/blob/eg-1269-basic-js-ws-confirmation/web/themes/weathergov_admin/weathergov_admin.theme#L109-L112), and is not unique to the dialog.
  
### Boring! How can I just try it out? ###
Here are some steps to follow to try it out locally:
* `zap` your containers
* Create a Weather Story for a specific WFO
* Publish the story
* Go back to the content page, and try to add a new Weather Story. Select the same WFO
* Click "Preview"
* Click the back link
* Click "Publish"
* A dialog should appear warning you about the original story being already published. Click "replace"
* You should see a message saying the new story was published and the old one was unpublished
* In the content list, there should only be one published weather story

## Screenshots (if appropriate): 📸
<details>
<summary>Visual summary of "replacing" with a new weather story for Nashville</summary>
  
Screenshot 2024-06-26 at 4 23 21 PM
Screenshot 2024-06-26 at 4 23 53 PM
Screenshot 2024-06-26 at 4 24 26 PM
Screenshot 2024-06-26 at 4 24 36 PM
Screenshot 2024-06-26 at 4 24 43 PM

</details>

### CODE REVIEW

File: web/themes/weathergov_admin/assets/js/weather-story-modal.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  // Add event handlers to the modal's buttons
+  Array.from(
+    modal.querySelectorAll("button"),
+  ).forEach((button) => {
+    button.addEventListener("click", () => {
+      // The data attribute will reference the id
+      // of a target button to programmatically click
+      // on the page's actual form.
+      // If there is no target, the modal will simply close
+      const targetButton = document.getElementById(
```
Comment (greg-does-weather) [inline web/themes/weathergov_admin/assets/js/weather-story-modal.js:85]: This is ***clever***.

**Review by greg-does-weather — CHANGES_REQUESTED** [review]
This is quite good. I know it's not what you originally set out to do, but this is nice. One comment that I think needs fixed, but once that's done, I think this is ready to ship and celebrate.

File: web/themes/weathergov_admin/weathergov_admin.theme [hunk 2]
```diff
@@ -95,10 +95,24 @@ function weathergov_admin_form_node_form_alter(
...
+            "weathergov_admin/weather-story-modal-dialog";
+    }
 }
 
 function weathergov_form_node_publish($form, &$formState)
 {
+    // First, archive any Weather Story nodes that are already
+    // published if they have the same WFO as was submitted
+    $wfo_id = $formState->getValue("field_wfo")[0]["target_id"];
+    weathergov_unpublish_published_weather_stories($wfo_id);
```
[the remark this answers is not in the document]
Comment (eric-gade) [reply web/themes/weathergov_admin/weathergov_admin.theme:112]: Whoa, oof. Good catch

[the remark this answers is not in the document]
Comment (eric-gade) [reply web/themes/weathergov_admin/weathergov_admin.theme]: Should be fixed now in https://github.com/weather-gov/weather.gov/pull/1392/commits/7cd1abaf56b281bdf5a4cca445f1553d19c15c0d thanks

**Review by greg-does-weather — APPROVED** [review]
Looks good

**Pull request merged**